### PR TITLE
Object mapper optimize

### PR DESCRIPTION
**Pull request by lixingwang**
**What kind of change does this PR introduce?** (check one with "x")
```
[] Bugfix
[] Feature
[] Code style update (formatting, local variables)
[] Refactoring (no functional changes, no api changes)
[*] Other... Please describe: Object mapper optimization because of performance issue.
```

**Fixes**: #

**What is the current behavior?**
Today we parse expression inside object mapper when running object mapper at runtime which causes performance bad

**What is the new behavior?**
Now we do parse expression inside expression at engine startup and runtime just execute it(Eval)

See benchmark improve.
```
Before
BenchmarkObj-12  100000	  17260 ns/op	   11648 B/op	  136 allocs/op

After:
BenchmarkObj-12   300000  3682 ns/op	    2472 B/op	   32 allocs/op
```

We can see which improve much better.

### CODE REVIEW

File: data/mapper/object.go [hunk 1]
```diff
@@ -83,182 +82,221 @@ func (am *ObjectMapperFactory) NewExpr(value string) (expression.Expr, error) {
...
+					}
+				}
+			}
+		}
+	}
+	return foreach, nil
+}
+
+func (obj *ObjectMapper) Eval(scope data.Scope) (value interface{}, err error) {
+	defer func() {
```
Comment (fm-tibco) [inline data/mapper/object.go]: Just a heads up that defer is expensive, we should make sure that we don't have too many of these nested

Comment (lixingwang) [reply data/mapper/object.go]: @fm-tibco  We have defer func at EvalActivity at TaskInst but we don't have defer func for trigger handler.   Do you think we should keep here or remove it from here and it at trigger handler?

Comment (fm-tibco) [reply data/mapper/object.go]: Lets move it to the trigger handler... this will help catch other issues as well.

Comment (lixingwang) [reply data/mapper/object.go]: Sure.  I removed defer here and will create another PR at flow repo to catch defer in the handler.

Comment (lixingwang) [reply data/mapper/object.go]: My fault.  I will add to this PR. Its part of core :)

**Pull request merged**